crates/node/engine/src/client.rs

```diff
@@ -59,106 +59,186 @@ impl EngineClient {
         let rpc_client = RpcClient::new(http_hyper, true);
         let engine = RootProvider::<AnyNetwork>::new(rpc_client);
 
-        let rpc = RootProvider::new_http(rpc);
-        let rpc = AlloyL2ChainProvider::new(rpc, cfg.clone());
+        let rpc = RootProvider::<Optimism>::new_http(rpc);
         Self { engine, rpc, cfg }
     }
 
-    /// Attempts to update the engine forkchoice state with the given attributes.
-    pub async fn forkchoice_updated_v2(
+    /// Fetches the [L2BlockInfo] by [BlockNumberOrTag].
+    pub async fn l2_block_info_by_label(
```

Comment:
Totally fine for this to be duplicated short-term, but we do have `L2BlockInfo::from_block_and_genesis` that we might be able to re-use here.